source/backend/dal/Repositories/ProductRepository.cs

```diff
@@ -42,11 +42,12 @@ public IList<PimsProduct> GetByProject(long projectId)
         /// </summary>
         /// <param name="incomingProducts"></param>
         /// <returns></returns>
-        public IEnumerable<PimsProduct> GetByProductBatch(IEnumerable<PimsProduct> incomingProducts)
+        public IEnumerable<PimsProduct> GetByProductBatch(IEnumerable<PimsProduct> incomingProducts, long projectId)
```

Comment:
maybe we need to change the name of the method or change the name of the parameter or add more documentation about this function. As it is, its a tad obscure what it is doing, and why it needs a project id

Reply:
fair, I'll add some more doc.